src/saltext/vmware/modules/vmc_distributed_firewall_rules.py

```diff
@@ -281,3 +292,461 @@ def delete(
+
+    Disabled
+        (Optional) Flag to disable the rule. Default is false.
+
+    logged
+        (Optional) Enable logging flag. Flag to enable packet logging. Default is disabled.
+
+    description
+        (Optional) Description of this resource
+
+    direction
+        (Optional) Define direction of traffic
+        Possible Values for are: IN, OUT, IN_OUT
+        Default: "IN_OUT"
+
+    notes
+        (Optional) Text for additional notes on changes.
+
+    tag
+        (Optional) Tag applied on the rule. User level field which will be printed in CLI and packet logs.
+
+    tags
+        (Optional) Opaque identifiers meaningful to the user.
+
+        .. code-block::
+
+            {
+                "tag": <tag>,
+                "scope": <scope>
+            }
+
+    Example Values:
+
+       .. code-block::
+
+           {
+              "description": " comm entry",
+              "sequence_number": 1,
+              "source_groups": [
+                  "ANY"
+              ],
+              "logged": false,
+              "destination_groups": [
+                  "ANY"
+              ],
+              "scope": [
+                  "ANY"
+              ],
+              "action": "DROP",
+              "services": [
+                  "ANY"
+              ],
+              "direction": "IN_OUT",
+              "tag": "",
+              "notes": ""
+            }
+
+    Notes:
+        For domain_id cgw
+        Rule must have valid values for source, and destination: either source or destination should have valid groups
+
+    Please refer the `Distributed firewall rules <https://developer.vmware.com/apis/nsx-vmc-policy/latest/policy/api/v1/infra/domains/domain-id/security-policies/security-policy-id/rules/rule-id/put/>`_ to get insight of input parameters
+
+    """
+    log.info("Creating Distributed firewall rule %s for SDDC %s", rule_id, sddc_id)
+
+    api_url_base = vmc_request.set_base_url(hostname)
+    api_url = (
+        "{base_url}vmc/reverse-proxy/api/orgs/{org_id}/sddcs/{sddc_id}/"
+        "policy/api/v1/infra/domains/{domain_id}/security-policies/{security_policy_id}/rules/{rule_id}"
+    )
+    api_url = api_url.format(
+        base_url=api_url_base,
+        org_id=org_id,
+        sddc_id=sddc_id,
+        domain_id=domain_id,
+        security_policy_id=security_policy_id,
+        rule_id=rule_id,
+    )
+
+    allowed_dict = {
+        "source_groups": source_groups,
+        "destination_groups": destination_groups,
+        "services": services,
+        "scope": scope,
+        "action": action,
+        "sequence_number": sequence_number,
+        "disabled": disabled,
+        "logged": logged,
+        "description": description,
+        "direction": direction,
+        "notes": notes,
+        "tag": tag,
+        "tags": tags,
+    }
+
+    req_data = vmc_request._filter_kwargs(
+        allowed_kwargs=allowed_dict.keys(), allow_none=["tags"], **allowed_dict
+    )
```

Comment:
**suggestion (non-blocking)** I haven't gone back to look at `_filter_kwargs` but it *seems* like this could be more like just an entry in `allowed_dict` :thinking:

Reply:
`allowed_dict` contains all the input fields (both mandatory and optional) needed for create API.
`_filter_kwargs` is to filter out the non-none values.